src/NFT/AgentCommunication.sol

```diff
@@ -46,39 +97,43 @@ contract AgentCommunication is Ownable {
         return (amountForTreasury, amountForAgent);
     }
 
+    // We don't add `onlyRegisteredAgent` modifier here, because anyone should be able to send a message to an agent
```

Comment:
Everything is guarded by `onlyRegisteredAgent`, except for this, because I guess we still want to be able to send messages from Streamlit UI to agents